### PR TITLE
Code coverage and new docker builder

### PR DESCRIPTION
**Pull request by y3ti**
Added a code coverage reports generated by `genhtml`.
Added a new docker builder, so it's easier to build and test locally without installing all required dependencies:
Just run

```
make clean
make build -e COVERAGE=1
make test
make coverage
```

PS. `CCACHE_DIR` was changed from `~/.ccache` into `.ccache`, so compile cache will start working after the merge.

### CODE REVIEW

**Review by cirochang — CHANGES_REQUESTED** [review]
Do we need to add gcovr as a dependency in `Depedencies` section of `README.md` ?

File: README.md [hunk 1]
```diff
@@ -35,10 +35,16 @@ make
...
 
 By default, these build steps will build the `oadr` and `oadrsd` libraries, and the
-`samplevenmanager` executable.  To build the unit test program `testoadr`, change the cmake
-command to the following:
+`samplevenmanager` executable. Before build the unit test program `testoadr`, I recommend to install the follow libraries:
 ```
-cmake -DTEST=1 ../../
+apt-get install --yes 
+    gcovr \
+    lcov
```
Comment (y3ti) [inline README.md]: We don't need `lcov`

Comment (y3ti) [reply README.md]: I see I added it to the `Dockerfile`. I will remove it.

Comment (y3ti) [reply README.md]: Ok, I removed `lcov` and I added how to run `gcovr`

File: Dockerfile [hunk 2]
```diff
@@ -1,24 +1,38 @@
...
+    &&  echo 'APT::Install-Suggests "false";'   > /etc/apt/apt.conf.d/00-no-suggestions \
+    &&  apt-get update \
+    &&  apt-get install --yes       \
+            ccache                  \
+            cmake                   \
+            g++                     \
+            gcovr                   \
+            libcurl4-gnutls-dev     \
+            libxerces-c-dev         \
+            libpthread-stubs0-dev   \
```
Comment (michal-kowalczyk) [inline Dockerfile:23]: why do we need `libpthread-stubs0-dev`?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+		$(DOCKER_EXTRA_ARGS) \
+		analyticsfire/openadr-ven-library-builder
+endef
+
+.PHONY: all
+all: build
+
+.PHONY: build
+build: builder
+	$(BUILDER) cmake -B build -DTEST=1 -DCOVERAGE=1
```
Comment (michal-kowalczyk) [inline Makefile]: It would be nice to have a binary produced which doesn't pay overload of coverage in runtime.

File: Dockerfile (hunk 2 above)
Comment (y3ti) [reply Dockerfile:23]: https://github.com/AnalyticsFire/OpenADR-VEN-Library/blob/master/CMakeLists.txt#L68

`samplevenmgr` requires `pthread` library.

File: Makefile (hunk 3 above)
Comment (y3ti) [reply Makefile]: I can enable code coverage by variable in Makefile e.g.,:

```
COVERAGE ?= 0
...
$(BUILDER) cmake -B build -DTEST=1 -DCOVERAGE=${COVERAGE}
```

and next we can run it via:

```
COVERAGE=1 make build
# or
make build -e COVERAGE=1
```

What do you think?

File: Dockerfile (hunk 2 above)
Comment (michal-kowalczyk) [reply Dockerfile:23]: Ah, right! Usually it is already installed on system but not sure if that's the case in docker.

File: Makefile (hunk 3 above)
Comment (michal-kowalczyk) [reply Makefile]: Yes, that looks great! When we publish build artefacts, we will most likely be interested in version COVERAGE=0.

**Review by cirochang — APPROVED** [review]
LGTM

File: Makefile (hunk 3 above)
Comment (y3ti) [reply Makefile]: Added in 54ced47

**Review by michal-kowalczyk — APPROVED** [review]

**Pull request merged**